api/api.go

```diff
@@ -119,35 +125,82 @@ func RequirePassword(h httprouter.Handle, password string) httprouter.Handle {
 	}
 }
 
+type cleanResponseWriter struct {
+	impl   http.ResponseWriter
+	m      sync.Mutex
+	closed bool
+}
+
+func (c *cleanResponseWriter) Close() {
+	c.m.Lock()
+	defer c.m.Unlock()
+	c.closed = true
+}
+
+func (c *cleanResponseWriter) Header() http.Header {
+	c.m.Lock()
+	defer c.m.Unlock()
+	if c.closed {
+		return http.Header{}
```

Comment:
I know that the server doesn't have a logger right now, but would this be something that we'd want to log? It's probably worth knowing both for devs and for users if something is calling functions on the response writer after it has been closed. Same for all the other methods that check the closed status